backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsync.java

```diff
@@ -0,0 +1,167 @@
+        Person newPatient =
+            new Person(
+                currentOrganization,
+                assignedFacility.orElse(null),
+                null, // lookupid
+                extractedData.getFirstName().getValue(),
+                extractedData.getMiddleName().getValue(),
+                extractedData.getLastName().getValue(),
+                extractedData.getSuffix().getValue(),
+                parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+                address,
+                country,
+                parsePersonRole(extractedData.getRole().getValue(), false),
+                List.of(extractedData.getEmail().getValue()),
+                convertRaceToDatabaseValue(extractedData.getRace().getValue()),
+                convertEthnicityToDatabaseValue(extractedData.getEthnicity().getValue()),
+                null, // tribalAffiliation
+                convertSexToDatabaseValue(extractedData.getBiologicalSex().getValue()),
+                parseYesNo(extractedData.getResidentCongregateSetting().getValue()),
+                parseYesNo(extractedData.getEmployedInHealthcare().getValue()),
+                null, // preferredLanguage
+                null // testResultDeliveryPreference
+                );
+
+        // only increment count after person record is created
+        patientCount++;
+
+        if (!patientsList.contains(newPatient)) {
+          // collect phone numbers and associate them with the patient
+          // then add to phone numbers list and set primary phone, if exists
+          List<PhoneNumber> newPhoneNumbers =
+              personService.assignPhoneNumbersToPatient(
+                  newPatient,
+                  List.of(
+                      new PhoneNumber(
+                          parsePhoneType(extractedData.getPhoneNumberType().getValue()),
+                          extractedData.getPhoneNumber().getValue())));
+          phoneNumbersList.addAll(newPhoneNumbers);
+          newPhoneNumbers.stream().findFirst().ifPresent(newPatient::setPrimaryPhone);
+
+          patientsList.add(newPatient);
+        }
+
+        if (patientCount >= batchSize) {
+          personService.addPatientsAndPhoneNumbers(patientsList, phoneNumbersList);
+          // clear lists after save, so we don't try to save duplicate records
+          patientsList.clear();
+          phoneNumbersList.clear();
+          patientCount = 0;
+        }
+
+      } catch (IllegalArgumentException e) {
+        String errorMessage = "Error uploading patient roster";
+        log.error(
+            errorMessage
+                + " for organization "
+                + currentOrganization.getExternalId()
+                + " and facility "
+                + facilityId);
+        throw new IllegalArgumentException(errorMessage);
+      }
+    }
+
+    personService.addPatientsAndPhoneNumbers(patientsList, phoneNumbersList);
```

Comment:
Is it ok that the batched calls to `addPatientsAndPhoneNumbers` are within a try block, but this one is not?

Reply:
I think it is, because that method is not currently configured to throw any exceptions anyway - I think it would just silently fail.